src/reducer.js

```diff
@@ -37,12 +39,17 @@ function myMerge(oldState, newState) {
         }
 
         const oldValue = result[key];
-        if (isFunction(oldValue.mergeDeep)) {
-            result[key] = oldValue.mergeDeep(value);
-        } else if (isFunction(value.mergeDeep)) {
-            result[key] = fromJS(oldValue).mergeDeep(value);
-        } else if (isObject(value) && !isArray(value)) {
-            result[key] = merge({}, oldValue, value);
+
+        if (!isNull(oldValue) && !isUndefined(oldValue)) {
```

Comment:
Isn't it enough to test `oldValue` to be truthy in the two `mergeDeep`-branches below (e.g. `(!!oldValue && isFunction(oldValue.mergeDeep))`)? 

I would prefer to not add two more dependencies for such a simple check. Or is there a special case I'm missing?